### ISSUE TITLE
Tx are delayed due to full blocks and incorrect fee calculation

### ISSUE THREAD

**Issue by ripper234** [OmniLayer/omniwallet#1256]
Bitcoin blocks have lately become saturated due to increased tx volume (see the whole Core/Classic debate).

This means that transactions generated by Omniwallet are sometimes stuck for a **long** time because they are paying lower fees that is needed to be included in a block. [See here](https://www.reddit.com/r/btc/comments/48kd7a/how_should_fees_be_calculated/) on how to correctly calculate fees under current conditions.

**Comment by ripper234** [comment]
> Wallets can't predict the future. There is no way to determine how many transaction will be added to miner's mempool and what fee will be attached to this transaction before next few blocks will get mined. In plain English - there is no way to determine future recommended fee (this is what wallet need to do to get fast confirmation). And, you know, some future block that will include your transaction - this block in the future. Any ideas how to resolve this? Besides something like 1 mBTC fee and blocksize limit increase (because seems like latter not going to happen anytime soon).

I think that some "sane heuristic" is probablly best here for now, to give us a bit of breathing room.

The ideal would be a dynamic formula that measures the fullness of latest blocks and predicts how much time on average it would take for a tx to go in one, based on these latest blocks. That's probablly? too much dev right now, so maybe start with a simply heuristic.

**Comment by msgilligan** [comment]
Bitcoin Core (and therefor Omni Core) have these calculations built-in and in the Qt interface accessible via a GUI slider. Since Omniwallet is now using Omni Core (mostly) it may be able to access these calculations via an RPC.

**Comment by zathras-crypto** [comment]
IIRC a little while back I walked Tomas through using the RPC interface
(estimatefee call) to recommend a fee to the user - perhaps this is only in
staging or pending further work.

Thanks
Z

**Comment by spacelite** [comment]
what if you implement a small omni % "tax" that only gets charged on
sending, and also builds into a "float" value, a hidden amount of btc from
that omni account that can reduce any future tax needs.

ps: Hi guys, long time no see. I just got soho set up.

**Comment by ripper234** [comment]
> ps: Hi guys, long time no see. I just got soho set up.

♥

**Comment by achamely** [comment]
Hey @ripper234 . Yes we do have plans to pull the estimatefee from omnicore rpc.  Right not just trying to get the staging rebrand finished merged. Once thats done we'll be able to move through a lot of other pending changes/updates/improvements.

**Comment by ripper234** [comment]
Thanks Adam
